### PR TITLE
Fix tracking for if a type has auto-layout fields for the managed type system.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
cc:@davidwrighton

### CODE REVIEW

File: src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs [hunk 1]
```diff
@@ -377,6 +377,7 @@ private static LayoutInt AlignUpInstanceFieldOffset(TypeDesc typeWithField, Layo
 
         protected ComputedInstanceFieldLayout ComputeSequentialFieldLayout(MetadataType type, int numInstanceFields)
         {
+            if (type.Name == "ObjectHandleOnStack") System.Diagnostics.Debugger.Break();
```
Comment (davidwrighton) [inline src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs:380]: Remove this.

**Review by davidwrighton — CHANGES_REQUESTED** [review]
Looks good except for the extra debugging code.

File: src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs:380]:
```suggestion
```

**Comment by jkoritzinsky** [comment]
Removed the debugging code

**Review by davidwrighton — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Only failure is #69476. Merging in

**Pull request merged**